app/services/ApiDefinitionService.scala

```diff
@@ -16,25 +16,21 @@
 
 package services
 
+import javax.inject.Inject
+
 import connectors.ApiDefinitionConnector
 import model.APIDefinition
 
 import scala.concurrent.Future
 import scala.concurrent.ExecutionContext.Implicits.global
 import uk.gov.hmrc.http.HeaderCarrier
 
-object ApiDefinitionService extends ApiDefinitionService {
-  override val apiDefinitionConnnector = ApiDefinitionConnector
-}
-
-trait ApiDefinitionService {
-
-  val apiDefinitionConnnector: ApiDefinitionConnector
+class ApiDefinitionService @Inject()(apiDefinitionConnector: ApiDefinitionConnector) {
 
   def fetchAllApiDefinitions(implicit hc: HeaderCarrier): Future[Seq[APIDefinition]] = {
     for {
-      publicApis <- apiDefinitionConnnector.fetchPublic()
-      privateApis <- apiDefinitionConnnector.fetchPrivate()
+      publicApis <- apiDefinitionConnector.fetchPublic()
```

Comment:
If you declare the futures then this will be completed in parallel rather than sequentially:

```
val publicApisFuture = apiDefinitionConnector.fetchPublic()
val privateApisFuture = apiDefinitionConnector.fetchPrivate()

for {
  privateApis <- privateApisFuture
  publicApis <- publicApisFuture
} yield (publicApis ++ privateApis).distinct
```